### DISCUSSION TITLE
I need to turn of MQTT on Rules

### DISCUSSION THREAD

**Discussion by Morphi08** [arendst/Tasmota#21861] [Q&A]
Hi there,
i have little trouble with MQTT and my rules; so I want to check up my Finnish "Waschmaschine" and send the Finnish over MQTT;
so I have 3 Rules for this:
`Rule1 on Energy#Power<3 do backlog var1 1; ruletimer1 300 endon on Energy#Power>=3 do var1 0 endon on Energy#Power>30 do backlog var1 0; ruletimer1 0 endon
`
`Rule2 on Rules#Timer=1 do if %var1%==1 publish stat/tasmota_E88366/POWER power below 3W endon
`
and `Rule3 on System#Boot do backlog Rule1 1; Rule2 1 endon
`

Now I have the problem, that my MQTT would be overcomes with Messages from the rules;
My Idea is, that I deactivated the mqtt for rules in the source-code...

Or have somebody any ideas, that no mqtt for my rules?

best regards

**Comment by barbudor** [comment]
> I have the problem, that my MQTT would be overcomes with Messages from the rules;

I'm not sure to understand what is your problem actually but I wouldn't use the POWER topic to publish the state of opération of your machine
I'm doing the same using OPERATING but that's a personnal choice

**Comment by barbudor** [comment]
For référence, here are my rules set:
```
18:18:31.885 MQT: stat/skt-washing/RULE = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":246,"Free":265,"Rules":"ON Energy#Power>%Mem1% DO IF (Var1==0) var9 %value%; Var1 1; Publish stat/%topic%/OPERATING ON ENDIF ENDON ON Energy#Power<=%Mem2% DO Backlog var9 %value%; RuleTimer1 %Mem3% ENDON ON Energy#Power>%Mem2% DO Backlog RuleTimer1 0; var9 %value% ENDON"}}
18:18:31.904 MQT: stat/skt-washing/RULE = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":211,"Free":300,"Rules":"ON Rules#Timer=1 DO IF (Var1==1) Var1 0; Publish stat/%topic%/OPERATING OFF ENDIF ENDON ON Time#Minute|%var2% DO IF (Var1==1) Publish tele/%topic%/OPERATING ON ELSE Publish tele/%topic%/OPERATING OFF ENDIF ENDON"}}
18:18:31.922 MQT: stat/skt-washing/RULE = {"Rule3":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":86,"Free":425,"Rules":"ON System#Boot DO backlog Teleperiod ENDON ON Teleperiod#data DO Var2=%value%/60 ENDON"}}

```

**Comment by Morphi08** [comment]
thx for your help;
my main problem is, I copied your rules on my tasmota and started, and then I have this in my console:
`17:46:22.591 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:22.794 MQT: stat/tasmota_E88366/RULETIMER = {"T1":300,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:23.164 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:23.196 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:23.398 MQT: stat/tasmota_E88366/RULETIMER = {"T1":300,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:23.765 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:23.797 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:24.048 MQT: stat/tasmota_E88366/RULETIMER = {"T1":299,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:24.366 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:24.397 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:24.649 MQT: stat/tasmota_E88366/RULETIMER = {"T1":299,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:24.967 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:24.998 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:25.251 MQT: stat/tasmota_E88366/RULETIMER = {"T1":300,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:25.571 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:25.601 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:25.804 MQT: stat/tasmota_E88366/RULETIMER = {"T1":300,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:26.173 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:26.204 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:26.456 MQT: stat/tasmota_E88366/RULETIMER = {"T1":299,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:26.773 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:26.804 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:27.056 MQT: stat/tasmota_E88366/RULETIMER = {"T1":299,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:46:27.377 RUL: ENERGY#POWER<=%MEM2% performs 'Backlog var9 2; RuleTimer1 300'
17:46:27.407 MQT: stat/tasmota_E88366/VAR = {"Var9":"2"}
17:46:27.660 MQT: stat/tasmota_E88366/RULETIMER = {"T1":300,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}`
and also on my matt-server ..in some minutes over 2700 messages..

this is my problem; do you know how to disable the "debugging" information for rule on console and mqtt?

**Reply by barbudor** [reply]
I've read in another thread about a feature recently added: if you add an underscore `_` at the beginning of a command, the MQTT outputs are disabled.
Also `Backlog3` does the same for all commands in the backlog

I haven't tried it by myself yet but may be @sfromis can confirm

**Reply by sfromis** [reply]
I'd say that the primary way of reducing volume would be to use one-shot mode to avoid the rule triggering repeatedly while the condition is still fulfilled without interruption.

Still, I can confirm that, as long as you use a quite recent development build of Tasmota, using `Backlog3` or prefixing the command name by `_` will indeed avoid the confirmation of command execution, but not all output.
